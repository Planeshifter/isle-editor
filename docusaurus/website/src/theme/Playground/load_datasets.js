import { json } from 'd3';

json( 'https://isle.heinz.cmu.edu/data-explorers_heartdisease.json' ).then( res => {
	global.heartdisease = res;
});
