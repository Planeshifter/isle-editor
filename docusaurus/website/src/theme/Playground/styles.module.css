/**
* Copyright (c) Facebook, Inc. and its affiliates.
*
* This source code is licensed under the MIT license found in the
* LICENSE file in the root directory of this source tree.
*/

.playgroundHeader {
	letter-spacing: 0.08rem;
	padding: 0.75rem;
	text-transform: uppercase;
	font-weight: bold;
}

.playgroundEditorHeader {
	background: var(--ifm-color-emphasis-600);
	color: var(--ifm-color-content-inverse);
}

.playgroundPreviewHeader {
	background: var(--ifm-color-emphasis-200);
	color: var(--ifm-color-content);
}

.playgroundEditor {
	font-family: var(--ifm-font-family-monospace) !important;
}

.playgroundPreview {
	border: 1px solid var(--ifm-color-emphasis-200);
	border-bottom-left-radius: var(--ifm-global-radius);
	border-bottom-right-radius: var(--ifm-global-radius);
	position: relative;
	padding: 1rem;
	transform: translateZ(0);
}

.btnCopy {
	float: right;
	z-index: 2;
	cursor: pointer;
	border: 1px solid transparent;
	padding: 0;
	color: white;
	background-color: transparent;
	transition: all 0.25s ease-out;
}

.btnCopy:hover {
	color: orange;
}
