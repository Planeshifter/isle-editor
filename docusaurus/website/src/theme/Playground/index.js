/* eslint-disable jsx-a11y/no-static-element-interactions */

/**
* Copyright (c) Facebook, Inc. and its affiliates.
*
* This source code is licensed under the MIT license found in the
* LICENSE file in the root directory of this source tree.
*/

// MODULES //

import * as React from 'react';
import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live';
import clsx from 'clsx';
import styles from './styles.module.css';
import './load_datasets.js';


// FUNCTIONS //

function preventPropagation( event ) {
	// Prevent propagation to avoid triggering Algolia when typing `/`:
	event.stopPropagation();
}


// MAIN //

function Playground({ children, theme, transformCode, ...props }) {
	return (
		<LiveProvider
			code={children.replace(/\n$/, '')}
			theme={theme}
			transformCode={( code ) => {
				return '<Provider session={session}><Lesson className="Lesson" >'+code+'</Lesson></Provider>;';
			}}
			{...props}>
			<div
				className={clsx(
					styles.playgroundHeader,
					styles.playgroundEditorHeader,
				)}>
				Live Editor
			</div>
			<LiveEditor
				className={styles.playgroundEditor}
				onKeyDown={preventPropagation}
			/>
			<div
				className={clsx(
					styles.playgroundHeader,
					styles.playgroundPreviewHeader,
				)}>
				Result
			</div>
			<div className={styles.playgroundPreview} onKeyDown={preventPropagation} >
				<LivePreview />
				<LiveError />
			</div>
		</LiveProvider>
	);
}


// EXPORTS //

export default Playground;
