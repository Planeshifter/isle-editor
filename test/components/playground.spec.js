// MODULES //

import React from 'react';
import renderer from 'react-test-renderer';
import Provider from '@isle-project/components/provider';
import Session from '@isle-project/session';
import Playground from 'editor-components/playground/main.js';
import Clock from '@isle-project/components/clock';


// VARIABLES //

const session = new Session( {}, true );


// TESTS //

it( 'renders a controlled playground', () => {
	const playground = renderer.create(
		<Provider session={session}>
			<Playground value="<div>Code comes here...</div>" />
		</Provider>
	).toJSON();
	expect( playground ).toMatchSnapshot();
});

it( 'renders an uncontrolled playground', () => {
	const playground = renderer.create(
		<Provider session={session}>
			<Playground defaultValue="<div>Code comes here...</div>" />
		</Provider>
	).toJSON();
	expect( playground ).toMatchSnapshot();
});

it( 'renders a playground with a supplied scope', () => {
	const playground = renderer.create(
		<Provider session={session}>
			<Playground defaultValue="<Clock />" scope={{
				Clock
			}} />
		</Provider>
	).toJSON();
	expect( playground ).toMatchSnapshot();
});
