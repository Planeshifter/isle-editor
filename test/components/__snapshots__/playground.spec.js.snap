// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a controlled playground 1`] = `null`;

exports[`renders a playground with a supplied scope 1`] = `
<div
  className="component-documentation"
  style={Object {}}
>
  <div
    className="playground-editable unselectable"
  >
    editable-source
  </div>
  <div
    className="playground-live-editor"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "color": "#e7d2ed",
          "fontFamily": "'Source Code Pro', monospace",
          "fontSize": "1.25em",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "left",
          "whiteSpace": "pre",
        }
      }
    >
      <textarea
        autoCapitalize="off"
        autoComplete="off"
        autoCorrect="off"
        className="npm__react-simple-code-editor__textarea"
        data-gramm={false}
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        spellCheck={false}
        style={
          Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "WebkitTextFillColor": "transparent",
            "background": "none",
            "border": 0,
            "boxSizing": "inherit",
            "color": "inherit",
            "display": "inherit",
            "fontFamily": "inherit",
            "fontSize": "inherit",
            "fontStyle": "inherit",
            "fontVariantLigatures": "inherit",
            "fontWeight": "inherit",
            "height": "100%",
            "left": 0,
            "letterSpacing": "inherit",
            "lineHeight": "inherit",
            "margin": 0,
            "overflow": "hidden",
            "overflowWrap": "break-word",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "position": "absolute",
            "resize": "none",
            "tabSize": "inherit",
            "textIndent": "inherit",
            "textRendering": "inherit",
            "textTransform": "inherit",
            "top": 0,
            "whiteSpace": "pre-wrap",
            "width": "100%",
            "wordBreak": "keep-all",
          }
        }
        value="<Clock />"
      />
      <pre
        aria-hidden="true"
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "inherit",
            "display": "inherit",
            "fontFamily": "inherit",
            "fontSize": "inherit",
            "fontStyle": "inherit",
            "fontVariantLigatures": "inherit",
            "fontWeight": "inherit",
            "letterSpacing": "inherit",
            "lineHeight": "inherit",
            "margin": 0,
            "overflowWrap": "break-word",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "pointerEvents": "none",
            "position": "relative",
            "tabSize": "inherit",
            "textIndent": "inherit",
            "textRendering": "inherit",
            "textTransform": "inherit",
            "whiteSpace": "pre-wrap",
            "wordBreak": "keep-all",
          }
        }
      >
        <div
          className="token-line"
          style={
            Object {
              "backgroundColor": null,
              "color": "#e7d2ed",
            }
          }
        >
          <span
            className="token tag punctuation"
            style={
              Object {
                "color": "#f677e1",
                "opacity": "0.7",
              }
            }
          >
            &lt;
          </span>
          <span
            className="token tag class-name"
            style={
              Object {
                "color": "#f677e1",
              }
            }
          >
            Clock
          </span>
          <span
            className="token tag"
            style={
              Object {
                "color": "#f677e1",
              }
            }
          >
             
          </span>
          <span
            className="token tag punctuation"
            style={
              Object {
                "color": "#f677e1",
                "opacity": "0.7",
              }
            }
          >
            /&gt;
          </span>
        </div>
      </pre>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "
/**
 * Reset the text fill color so that placeholder is visible
 */
.npm__react-simple-code-editor__textarea:empty {
  -webkit-text-fill-color: inherit !important;
}

/**
 * Hack to apply on some CSS on IE10 and IE11
 */
@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
  /**
    * IE doesn't support '-webkit-text-fill-color'
    * So we use 'color: transparent' to make the text transparent on IE
    * Unlike other browsers, it doesn't affect caret color in IE
    */
  .npm__react-simple-code-editor__textarea {
    color: transparent !important;
  }

  .npm__react-simple-code-editor__textarea::selection {
    background-color: #accef7 !important;
    color: transparent !important;
  }
}
",
          }
        }
        type="text/css"
      />
    </div>
  </div>
  <div
    className="playground-live-preview"
    style={
      Object {
        "height": "calc(100% - 33px)",
        "overflowX": "hidden",
        "overflowY": "hidden",
        "position": "absolute",
        "right": 0,
        "top": 33,
        "transform": "translateZ(0)",
        "width": "50%",
      }
    }
  />
  <pre
    style={
      Object {
        "background": "#F2777A",
        "color": "white",
        "fontSize": "1.1em",
        "padding": "10px",
        "position": "absolute",
        "right": 0,
        "top": 33,
        "width": "50%",
        "zIndex": "9999",
      }
    }
  >
    ReferenceError: Provider is not defined
  </pre>
</div>
`;

exports[`renders an uncontrolled playground 1`] = `
<div
  className="component-documentation"
  style={Object {}}
>
  <div
    className="playground-editable unselectable"
  >
    editable-source
  </div>
  <div
    className="playground-live-editor"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "color": "#e7d2ed",
          "fontFamily": "'Source Code Pro', monospace",
          "fontSize": "1.25em",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "left",
          "whiteSpace": "pre",
        }
      }
    >
      <textarea
        autoCapitalize="off"
        autoComplete="off"
        autoCorrect="off"
        className="npm__react-simple-code-editor__textarea"
        data-gramm={false}
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        spellCheck={false}
        style={
          Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "WebkitTextFillColor": "transparent",
            "background": "none",
            "border": 0,
            "boxSizing": "inherit",
            "color": "inherit",
            "display": "inherit",
            "fontFamily": "inherit",
            "fontSize": "inherit",
            "fontStyle": "inherit",
            "fontVariantLigatures": "inherit",
            "fontWeight": "inherit",
            "height": "100%",
            "left": 0,
            "letterSpacing": "inherit",
            "lineHeight": "inherit",
            "margin": 0,
            "overflow": "hidden",
            "overflowWrap": "break-word",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "position": "absolute",
            "resize": "none",
            "tabSize": "inherit",
            "textIndent": "inherit",
            "textRendering": "inherit",
            "textTransform": "inherit",
            "top": 0,
            "whiteSpace": "pre-wrap",
            "width": "100%",
            "wordBreak": "keep-all",
          }
        }
        value="<div>Code comes here...</div>"
      />
      <pre
        aria-hidden="true"
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "inherit",
            "display": "inherit",
            "fontFamily": "inherit",
            "fontSize": "inherit",
            "fontStyle": "inherit",
            "fontVariantLigatures": "inherit",
            "fontWeight": "inherit",
            "letterSpacing": "inherit",
            "lineHeight": "inherit",
            "margin": 0,
            "overflowWrap": "break-word",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "pointerEvents": "none",
            "position": "relative",
            "tabSize": "inherit",
            "textIndent": "inherit",
            "textRendering": "inherit",
            "textTransform": "inherit",
            "whiteSpace": "pre-wrap",
            "wordBreak": "keep-all",
          }
        }
      >
        <div
          className="token-line"
          style={
            Object {
              "backgroundColor": null,
              "color": "#e7d2ed",
            }
          }
        >
          <span
            className="token tag punctuation"
            style={
              Object {
                "color": "#f677e1",
                "opacity": "0.7",
              }
            }
          >
            &lt;
          </span>
          <span
            className="token tag"
            style={
              Object {
                "color": "#f677e1",
              }
            }
          >
            div
          </span>
          <span
            className="token tag punctuation"
            style={
              Object {
                "color": "#f677e1",
                "opacity": "0.7",
              }
            }
          >
            &gt;
          </span>
          <span
            className="token plain"
          >
            Code comes here
          </span>
          <span
            className="token operator"
            style={
              Object {
                "color": "hsl(40, 90%, 70%)",
              }
            }
          >
            ...
          </span>
          <span
            className="token tag punctuation"
            style={
              Object {
                "color": "#f677e1",
                "opacity": "0.7",
              }
            }
          >
            &lt;/
          </span>
          <span
            className="token tag"
            style={
              Object {
                "color": "#f677e1",
              }
            }
          >
            div
          </span>
          <span
            className="token tag punctuation"
            style={
              Object {
                "color": "#f677e1",
                "opacity": "0.7",
              }
            }
          >
            &gt;
          </span>
        </div>
      </pre>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "
/**
 * Reset the text fill color so that placeholder is visible
 */
.npm__react-simple-code-editor__textarea:empty {
  -webkit-text-fill-color: inherit !important;
}

/**
 * Hack to apply on some CSS on IE10 and IE11
 */
@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
  /**
    * IE doesn't support '-webkit-text-fill-color'
    * So we use 'color: transparent' to make the text transparent on IE
    * Unlike other browsers, it doesn't affect caret color in IE
    */
  .npm__react-simple-code-editor__textarea {
    color: transparent !important;
  }

  .npm__react-simple-code-editor__textarea::selection {
    background-color: #accef7 !important;
    color: transparent !important;
  }
}
",
          }
        }
        type="text/css"
      />
    </div>
  </div>
  <div
    className="playground-live-preview"
    style={
      Object {
        "height": "calc(100% - 33px)",
        "overflowX": "hidden",
        "overflowY": "hidden",
        "position": "absolute",
        "right": 0,
        "top": 33,
        "transform": "translateZ(0)",
        "width": "50%",
      }
    }
  />
  <pre
    style={
      Object {
        "background": "#F2777A",
        "color": "white",
        "fontSize": "1.1em",
        "padding": "10px",
        "position": "absolute",
        "right": 0,
        "top": 33,
        "width": "50%",
        "zIndex": "9999",
      }
    }
  >
    ReferenceError: Provider is not defined
  </pre>
</div>
`;
